openrr-planner/src/collision/self_collision_checker.rs

```diff
@@ -48,14 +57,15 @@ impl SelfCollisionChecker {
 
     pub fn check_joint_positions(
         &self,
-        current: &[f64],
-        positions: &[f64],
+        current: &[N],
+        positions: &[N],
         duration: std::time::Duration,
     ) -> Result<()> {
         match interpolate(
             &[current.to_vec(), positions.to_vec()],
-            duration.as_secs_f64(),
-            duration.as_secs_f64() * self.time_interpolate_rate,
+            num_traits::NumCast::from::<f64>(duration.as_secs_f64()).unwrap(),
```

Comment:
How about using variable to reduce the twice complicated conversion.

```rust
let duration_f64: f64 = num_traits::NumCast::from::<f64>(duration.as_secs_f64()).unwrap();
```

And aren't there any more easy conversions?

Reply:
Thanks, I accept your suggestion introducing a new variable.
Unfortunately, I do not know nor find any better conversion.